### PR TITLE
add test_utils

### PR DESCRIPTION
**Pull request by HerrMuellerluedenscheid**
Note the changed logging style in line 44 of `utils.py`. Formatting a `logging` string should be handled by the logging formatter: https://docs.python.org/3/library/logging.html#formatter-objects

We may consider using another formatting style but that should be consistent throught the entire application: https://docs.python.org/3/howto/logging-cookbook.html#formatting-styles

That is contradictory to what `pyupgrade` defaults to. I haven't checked if there is a flag that let's `pyupgrade` switch to `%s` for logging.

### CODE REVIEW

File: amqtt/utils.py [hunk 1]
```diff
@@ -5,12 +5,17 @@
...
 
+if typing.TYPE_CHECKING:
+    from amqtt.session import Session
 
 logger = logging.getLogger(__name__)
 
 
-def format_client_message(session=None, address=None, port=None):
+def format_client_message(
+    session: "Session" = None, address: str = None, port: int = None
```
Comment (FlorianLudwig) [inline amqtt/utils.py:17]: personal preference: Using `from __future__ import annotations` instead of directly using string

**Review by FlorianLudwig — APPROVED** [review]
First of all, sorry for the late review.

I like the addition of hypothesis.

My comment is nit picking / opinion so feel free to merge :)

File: amqtt/utils.py (hunk 1 above)
Comment (HerrMuellerluedenscheid) [reply amqtt/utils.py]: Yeah, right! A little less mixture of styles. Can you please have a second look? If you approve, feel free to merge.

Comment (HerrMuellerluedenscheid) [reply amqtt/utils.py]: It that `python3.7+` only? Shell we deprecate `python3.6` support?

**Pull request merged**